src/pkg/services/m365/api/lists.go

```diff
@@ -472,15 +490,45 @@ func setAdditionalDataByColumnNames(
 	fieldData := orig.GetAdditionalData()
 	filteredData := make(map[string]any)
 
-	for colName := range columnNames {
-		if _, ok := fieldData[colName]; ok {
+	for colName, colDetails := range columnNames {
+		if val, ok := fieldData[colName]; ok {
+			setMultipleEnabledByFieldData(val, colDetails)
+
 			filteredData[colName] = fieldData[colName]
 		}
+
+		specifyODataType(filteredData, colDetails, colName)
 	}
 
 	return filteredData
 }
 
+func setMultipleEnabledByFieldData(val any, colDetails *columnDetails) {
+	// for columns like 'choice', even though it has an option to hold single/multiple values,
+	// the columnDefinition property 'allowMultipleValues' is not available.
+	// Hence we determine single/multiple from the actual field data.
+	if reflect.TypeOf(val).Kind() == reflect.Slice {
+		colDetails.isMultipleEnabled = true
+	}
+}
```

Comment:
Preference on general func behavior: rather than setting values a provided object (which causes coupling between the caller and the func), funcs that provide return values and let callers act on them (which provides cohesion without coupling), are easier to maintain in the long term.  In this case, it's a better pattern to have a func whose only job is to check whether the value is a slice.  Ex:
```suggestion
func isSlice(val any) bool {
	return reflect.TypeOf(val).Kind() == reflect.Slice
}
```